service/k8sclient/kubectl/kubectl.go

```diff
@@ -0,0 +1,130 @@
+// dbaas-controller
+// Copyright (C) 2020 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+// Package kubectl provides kubectl CLI wrapper.
+package kubectl
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"os/exec"
+	"strings"
+
+	"github.com/percona/pmm/utils/pdeathsig"
+	"github.com/pkg/errors"
+	"golang.org/x/sys/unix"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/percona-platform/dbaas-controller/utils/logger"
+)
+
+// errNotFound is an error in case of object not found.
+var errNotFound = errors.New("object not found")
+
+// KubeCtl wraps kubectl CLI with version selection and kubeconfig handling.
+type KubeCtl struct {
+	l   logger.Logger
+	cmd []string
+}
+
+// NewKubeCtl creates a new KubeCtl object with a given logger.
+func NewKubeCtl(l logger.Logger) *KubeCtl {
+	// TODO Handle kubectl versions https://jira.percona.com/browse/PMM-6348
+
+	// TODO Handle kubeconfig https://jira.percona.com/browse/PMM-6347
+
+	cmd := []string{"dbaas-kubectl-1.16"}
+	path, err := exec.LookPath(cmd[0])
+	l.Debugf("path: %s, err: %v", path, err)
+	if e, ok := err.(*exec.Error); err != nil && ok && e.Err == exec.ErrNotFound {
+		cmd = []string{"minikube", "kubectl", "--"}
+	}
+	l.Debugf("Using %q", strings.Join(cmd, " "))
+
+	return &KubeCtl{
+		l:   l.WithField("component", "kubectl"),
+		cmd: cmd,
+	}
+}
+
+// Cleanup removes temporary files created by that object.
+func (k *KubeCtl) Cleanup() {
+	// TODO Remove kubeconfig file https://jira.percona.com/browse/PMM-6347
+}
+
+// Get executes `kubectl get` with given object kind and optional name,
+// and decodes resource into `res`.
+// It returns `errNotFound` if object is not found.
+func (k *KubeCtl) Get(ctx context.Context, kind string, name string, res interface{}) error {
+	args := []string{"get", "-o=json", kind}
+	if name != "" {
+		args = append(args, name)
+	}
+
+	stdout, stderr, err := k.run(ctx, args, nil)
+	if err != nil {
+		if strings.Contains(string(stderr), "not found") { // FIXME
```

Comment:
Can we do better?

Reply:
Removed it since we don't need it anymore.